### ISSUE TITLE
The output shape of torch.nn.quantized.functional.max_pool2d() is wrong in some cases

### ISSUE THREAD

**Issue by wbn520** [pytorch/pytorch#47556]
## To Reproduce

When the kenel_size is set to (2,2) and padding is (1,1), the output shape of  torch.nn.quantized.functional.max_pool2d() is different from the result based on the formula in the doc.

## Expected behavior

Such as if the parameters are set as follow:
ksize = (2,2)
stride = (2,2)
padding = (1,1)
dilation = 1
The input tensor shape is (2,3,11,9)
the expected outputshape is (2,3,7,6) based on the formula.
But the real result is (2,3,6,5), 
And whether the ceil_mode is False or True, The result is same.

## Environment
python version = 3.7
torch version = 1.6.0

```
import torch
import numpy as np

def create_tensor(shape, dtype, min=0, max=100):
    data = np.random.uniform(low=min,high=max,size=shape).astype(dtype)
    tensor = torch.from_numpy(data)
    return tensor

shape = (2,3,11,9)
tensor = create_tensor(shape, np.float32, 0, 100)
ksize = (2,2)
stride = (2,2)
padding = (1,1)
dilation = 1
ceil_mode = False
output1 = torch.nn.quantized.functional.max_pool2d(tensor, ksize, stride, padding, dilation, ceil_mode)
ceil_mode = True
output2 = torch.nn.quantized.functional.max_pool2d(tensor, ksize, stride, padding, dilation, ceil_mode)
print(output1.shape)
print(output2.shape)
```

## Additional context

Is this a bug or is it that I understand it wrong？

**Comment by heitorschueroff** [comment]
@wbn520 Please, see the following PR which will update the documentations to reflect the current behavior https://github.com/pytorch/pytorch/pull/46558/files.

TLDLR: The equation includes right padding but the code does not allow a sliding window to start within the right-padded region. In this case, the last sliding window would start in the right padding which is why you see the different of 1 in the output size. We may change this in the future to include the right padding, but for now we are only updating the documentation to reflect this.
